hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -116,8 +117,25 @@ public HddsDatanodeService(boolean printBanner, String[] args) {
     this.args = args != null ? Arrays.copyOf(args, args.length) : null;
   }
 
+  private void cleanTmpDir() {
+    if (datanodeStateMachine != null) {
+      MutableVolumeSet volumeSet =
+          datanodeStateMachine.getContainer().getVolumeSet();
+      for (HddsVolume hddsVolume : StorageVolumeUtil.getHddsVolumesList(
```

Comment:
@neils-dev In `ScmHAFinalizeUpgradeActionDatanode` we are iterating `volumeSet.getVolumesList()` which returns all volumes and in that case, it makes sense to check their instance. 

Here, we are wraping the `volumeSet.getVolumesList()` with `StorageVolumeUtil.getHddsVolumesList(volume list)` which filters the volumes and returns only the HddsVolumes.

https://github.com/apache/ozone/blob/master/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/StorageVolumeUtil.java#L66

Reply:
Thanks @xBis7 , that util, it seems to _cast_ each entry in the volume stream to an `HddsVolume` and returns the corresponding list.  It would be better to iterate the volume set and only use `HddsVolume`.

Reply:
I didn't realize it was casting to `HddsVolume`. You are right, in that case, it's best to iterate all volumes and check the instance. I'll fix it.